internal/generator/vector/apiaudit/helpers_test.go

```diff
@@ -0,0 +1,113 @@
+package apiaudit
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/cluster-logging-operator/test"
+	"k8s.io/apimachinery/pkg/runtime"
+	. "k8s.io/apiserver/pkg/apis/audit/v1"
+	"sigs.k8s.io/yaml"
+)
+
+const vectorImage = "quay.io/openshift-logging/vector:5.8.0"
+
+func Filtered(policy *AuditPolicy, event Event) *Event {
+	cmd := vectorCmd(policy)
+	b, err := json.Marshal(event)
+	test.Must(err)
+	cmd.Stdin = bytes.NewReader(b)
+	cmd.Stderr = GinkgoWriter
+	out, err := cmd.Output()
+	test.Must(err)
+	if strings.TrimSpace(string(out)) == "" {
+		return nil
+	}
+	e2 := &Event{}
+	test.Must(json.Unmarshal(out, e2))
+	return e2
+}
+
+// vectorCmd creates a temp dir for vector to use and returns the exec.Cmd
+func vectorCmd(p *AuditPolicy) *exec.Cmd {
+	vrl, err := PolicyToVRL(p)
+	Expect(err).NotTo(HaveOccurred(), "%#v", *p)
+	dir, err := os.MkdirTemp("", "apiaudit_test")
+	GinkgoT().Cleanup(func() { _ = os.RemoveAll(dir) })
+	Expect(err).NotTo(HaveOccurred())
+	Expect(os.WriteFile(filepath.Join(dir, "vector.toml"), []byte(vectorConf), 0666)).To(Succeed())
+	Expect(os.WriteFile(filepath.Join(dir, "audit_policy.vrl"), []byte(vrl), 0666)).To(Succeed())
+	return exec.Command("podman", "run", "-i", "--rm", "--env", "VECTOR_LOG=warn", "-v", dir+":/etc/vector:Z", vectorImage)
```

Comment:
Do we really want to bind our unit tests to podman?  I'm fairly certain podman is not available on CI.  We may be better served either making some flavor of a functional test or figuring out how to prep CI to download vector  and run these.  Maybe  a special Docker image we build in CI which combines vector and adds the source?

Reply:
Will convert to functional unless I can think of a better solution, which I can't so far.